### PR TITLE
[#154] Implementing People Card (SearchUserDetails) Component

### LINKED ISSUE #154: Implementing People Card (SearchUserDetails) Component
**Is your feature request related to a problem? Please describe.**
When searching for a new user, a people card component needs to be designed that showcases the information of the appropriate user(s). 

**Describe the solution you'd like**
The People card component would have a design specified in the Figma mockups. This component would display the relevant user's name, handle, bio and provides an option to follow or unfollow the user. 

**Describe alternatives you've considered**
N/A

**Additional context**

- This issue was brought to light, approved and allocated  in the frontend team meeting on the 24th March 2022
- Here is a mockup of the People card component.

image

### PR DESCRIPTION
**Pull request by the1e**
# Description:
Fixes #154. Added a `PeopleCard` component _(renamed it to `SearchUserDetails` component to maintain consistency with the previous components in `src/components/user/`)_. 

The SearchUserDetails component displays any resulting user profiles while the user is performing a search. Media queries have been used to make the component responsive to all devices. 

A `FollowButton` component has also been added as a part of the SearchUserDetails component. When clicked on, the component changes the Button text from following to follow or vice versa. It also has an `onClick` function that would be called when the user clicks to "Follow" or "UnFollow" someone.

_Update_: Implemented View and Controller design 

<hr>

Screenshot of how the `SearchUserDetails` component and `FollowButton` component look have been provided below.

<p align="center">
  image
image
</p>

An example of how `SearchUserDetails` component can be used is as follows:

```javascript
 <SearchUserDetails
  username= "Prison Mike",
  handle= "littlekidlover",  
  bio= "The worst thing about prison was the... was the dementors. They were flying all over the place, and they were scary and then theyd come down and theyd suck the soul out of your body. And it hurt!",
  profilePic="https://cdn.costumewall.com/wp-content/uploads/2018/09/prison-mike.jpg"
/>
```
An example of how `FollowButton` component can be used is as follows:

```javascript
 <FollowButton 
    isFollowing={true}; 
    onClick={() => {}}  
  />
```
<hr>
### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](../CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My code follows the code style of this project
- [x] My code has been commented
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

**Review by shreyasingaraju — APPROVED** [review]
Looks good to me! Great work :) 

Perhaps we can make a Buttons folder under Components since the follow button will be re-used on multiple pages?

**Comment by dshah1901** [comment]
> 

Okay, made another buttons folder in components :) Thank you!

**Comment by the1e** [comment]
> Looks good to me! Great work :)
> 
> Perhaps we can make a Buttons folder under Components since the follow button will be re-used on multiple pages?


Okay, made another buttons folder in components :) Thank you!

**Review by arathibiju — APPROVED** [review]
LGTM! great work on the follow button

**Review by shreyasingaraju — APPROVED** [review]
Structure looks great with the View and Controller design. Also looks cleaner with the button component added separately. Looks good to merge! 👍

**Review by arathibiju — APPROVED** [review]
The controller view format looks good!

File: frontend/src/components/button/followbutton/followbutton.module.scss [hunk 1]
```diff
@@ -0,0 +1,20 @@
+@use '~/src/styles/theme';
+@import '~/src/styles/theme';
+
+.followButton {
+  // Had to use !important tag as these properties were overridden by default css of Button MUI
```
Comment (uoa-lzha989) [inline frontend/src/components/button/followbutton/followbutton.module.scss:5]: This is interesting - just took a look and it seems like using !important has a "use as a last resort" attached to it (?) - do none of these options work on it? https://mui.com/customization/how-to-customize/

File: frontend/src/components/user/searchdetails/SearchUserDetailsView.jsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+/**
+ * The SearchUserDetailsView component is used to display the any results for users while performing a search.
+ *
+ * Props:
+ * @username : the username of the appropriate user.
+ * @handle : the handle (@) of the appropriate user.
+ * @bio : the biography of the appropriate user.
+ * @profilePic : the link to the profile image of the appropriate user.
+ *
+ * The FollowButton function has two props isFollowing and OnClick which are can be change to appropriate values when needed.
```
Comment (uoa-lzha989) [inline frontend/src/components/user/searchdetails/SearchUserDetailsView.jsx]: Might need to say "... and OnClick which can be changed to appropriate values..."

**Review by uoa-lzha989 — COMMENTED** [review]
It's all looking really good to me, just a few things to check on my end but otherwise good job!

**Review by ppat965 — APPROVED** [review]
The component final view looks really good and professional

File: frontend/src/components/button/followbutton/followbutton.module.scss (hunk 1 above)
Comment (the1e) [reply frontend/src/components/button/followbutton/followbutton.module.scss:5]: Ok so here is the problem. The classes and global overriding, I tried it but doesn't work, (used .Mui-Button-root) but the button remains the same. 

Now the things that worked for me are: !important and [Reusable styles overrides.](https://mui.com/customization/how-to-customize/#2-reusable-style-overrides). 
The problem with reusable styles and inline styling, I cannot use the `theme.scss` provided by the team and had to use hex or rgb codes. 

Is there any other way I can do this?

Hope this makes sense

File: frontend/src/components/user/searchdetails/SearchUserDetailsView.jsx (hunk 2 above)
Comment (the1e) [reply frontend/src/components/user/searchdetails/SearchUserDetailsView.jsx]: My brain at 1 am doesn't know English haha. Thank you for pointing it out :)

**Review by shreytailor — APPROVED** [review]
LGTM! Great work

**Review by scorpionknifes — APPROVED** [review]

**Pull request merged**